eventdata/challenges/bulk-size-evaluation.json

```diff
@@ -114,89 +130,53 @@
   "meta": {
     "benchmark_type": "indexing_bulksize"
   },
   "schedule": [
     {
-      "name": "delete-index-2clients",
-      "operation": "delete-index"
-    },
-    {
-      "name": "create-index-2clients",
-      "operation": {
-        "operation-type": "create-index",
-        "settings": {
-            "index.number_of_replicas": {{ p_replica_count }},
-            "index.number_of_shards": {{ p_shard_count }}
-        }
-      }
-    },
-    {
-      "name": "index-append-1000-2clients",
-      "operation": {
-        "operation-type": "bulk",
-        "param-source": "elasticlogs_bulk",
-        "bulk-size": 1000,
-        "record_raw_event_size": {{p_record_raw_event_size}}
-      },
-      "warmup-time-period": 300,
-      "time-period": 600,
-      "clients": 2
-    },
-    {
-      "name": "delete-index-20clients",
-      "operation": "delete-index"
+      "operation": "delete-index-template"
     },
     {
-      "name": "create-index-20clients",
-      "operation": {
-        "operation-type": "create-index",
-        "settings": {
-            "index.number_of_replicas": {{ p_replica_count }},
-            "index.number_of_shards": {{ p_shard_count }}
-        }
-      }
+      "operation": "create-index-template"
     },
+{% set comma = joiner() %}
+{% for factor in [1, 10, 50] %}
+{{comma()}}
+{% set concurrency_evaluation_clients = 2 * factor %}
+{% set concurrency_evaluation_bulk_size = 1000 // factor %}
     {
-      "name": "index-append-100-20clients",
-      "operation": {
-        "operation-type": "bulk",
-        "param-source": "elasticlogs_bulk",
-        "bulk-size": 100,
-        "record_raw_event_size": {{p_record_raw_event_size}}
-      },
-      "warmup-time-period": 300,
-      "time-period": 600,
-      "clients": 20
-    },
-    {
-      "name": "delete-index-200clients",
-      "operation": "delete-index"
+      "name": "delete-index-{{concurrency_evaluation_clients}}clients",
+      "operation": "deleteindex_elasticlogs"
     },
     {
-      "name": "create-index-200clients",
+      "name": "create-index-{{concurrency_evaluation_clients}}clients",
       "operation": {
         "operation-type": "create-index",
-        "settings": {
-            "index.number_of_replicas": {{ p_replica_count }},
-            "index.number_of_shards": {{ p_shard_count }}
+        "index": "elasticlogs",
+        "body": {
+          "settings": {
+              "index.number_of_replicas": {{ p_replica_count }},
+              "index.number_of_shards": {{ p_shard_count }}
+          }
         }
       }
     },
     {
-      "name": "index-append-10-200clients",
+      "name": "index-append-{{concurrency_evaluation_bulk_size}}-{{concurrency_evaluation_clients}}clients",
       "operation": {
         "operation-type": "bulk",
         "param-source": "elasticlogs_bulk",
-        "bulk-size": 10,
```

Comment:
AFAIU the j2 loop above will use [1, 10, 50] and since we set `set concurrency_evaluation_bulk_size = 1000 // factor` this will result in bulk sizes of 1000, 100 and **20**; the old setting in this line uses bulk-size **10**; is this change on purpose?

Reply:
Yes, this is indeed on purpose because (at least on my machine) handling 100 clients was ok whereas 200 was not (it led to connection drops in the actor system)

Reply:
Thanks SGTM